client/service_create.go

```diff
@@ -18,13 +18,25 @@ func (cli *Client) ServiceCreate(ctx context.Context, service swarm.ServiceSpec,
 		}
 	}
 
+	// Call the /manifest endpoint to retrieve manifest information
+	manifestInspect, manifestErr := cli.ImageManifest(ctx, service.TaskTemplate.ContainerSpec.Image, options.EncodedRegistryAuth)
+	if manifestErr == nil {
+		// manifest was successfully retrieved, now pin by digest
+		service.TaskTemplate.ContainerSpec.Image = string(manifestInspect.Digest)
```

Comment:
We must not overwrite the `Image` field here if it already contains a digest (for example, if digest pinning was already done through content trust).